fvm/evm/evm_test.go

```diff
@@ -223,32 +433,148 @@ func TestCadenceOwnedAccountDeploy(t *testing.T) {
 					let cadenceOwnedAccount <- EVM.createCadenceOwnedAccount()
 					cadenceOwnedAccount.deposit(from: <-vault)
 
-					let address = cadenceOwnedAccount.deploy(
-						code: [],
-						gasLimit: 53000,
-						value: EVM.Balance(attoflow: 1230000000000000000)
+					let bal = EVM.Balance(0)
+					bal.setFLOW(flow: 1.23)
+
+					let recipientEVMAddress = EVM.EVMAddress(bytes: address)
+
+					let res = cadenceOwnedAccount.call(
+						to: recipientEVMAddress,
+						data: [],
+						gasLimit: 100_000,
+						value: bal,
 					)
+
+					assert(res.status == EVM.Status.successful, message: "transfer call was not successful")
+
 					destroy cadenceOwnedAccount
-					return address.bytes
+					return recipientEVMAddress.balance().inFLOW()
 				}
-                `,
-				sc.EVMContract.Address.HexWithPrefix(),
-				sc.FlowToken.Address.HexWithPrefix(),
-				sc.FlowServiceAccount.Address.HexWithPrefix(),
-			))
+				`,
+					sc.EVMContract.Address.HexWithPrefix(),
+					sc.FlowToken.Address.HexWithPrefix(),
+					sc.FlowServiceAccount.Address.HexWithPrefix(),
+				))
 
-			script := fvm.Script(code)
+				addr := cadence.NewArray(
+					ConvertToCadence(testutils.RandomAddress(t).Bytes()),
+				).WithType(stdlib.EVMAddressBytesCadenceType)
 
-			executionSnapshot, output, err := vm.Run(
-				ctx,
-				script,
-				snapshot)
-			require.NoError(t, err)
-			require.NoError(t, output.Err)
+				script := fvm.Script(code).WithArguments(
+					json.MustEncode(addr),
+				)
 
-			// TODO:
-			_ = executionSnapshot
-		})
+				_, output, err := vm.Run(
+					ctx,
+					script,
+					snapshot)
+				require.NoError(t, err)
+				require.NoError(t, output.Err)
+
+				require.Equal(t, uint64(123000000), uint64(output.Value.(cadence.UFix64)))
+			})
+	})
+
+	t.Run("test coa deposit and withdraw in a single transaction", func(t *testing.T) {
+		RunWithNewEnvironment(t,
+			chain, func(
+				ctx fvm.Context,
+				vm fvm.VM,
+				snapshot snapshot.SnapshotTree,
+				testContract *TestContract,
+				testAccount *EOATestAccount,
+			) {
+				code := []byte(fmt.Sprintf(
+					`
+				import EVM from %s
+				import FlowToken from %s
+
+				access(all)
+				fun main(): UFix64 {
+					let admin = getAuthAccount(%s)
+						.borrow<&FlowToken.Administrator>(from: /storage/flowTokenAdmin)!
+					let minter <- admin.createNewMinter(allowedAmount: 2.34)
+					let vault <- minter.mintTokens(amount: 2.34)
+					destroy minter
+
+					let cadenceOwnedAccount <- EVM.createCadenceOwnedAccount()
+					cadenceOwnedAccount.deposit(from: <-vault)
+
+					let bal = EVM.Balance(0)
+					bal.setFLOW(flow: 1.23)
+					let vault2 <- cadenceOwnedAccount.withdraw(balance: bal)
```

Comment:
nit: maybe you can add a line in Cadence that asserts the coa balance is `2.34-1.23`.